core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -465,6 +465,8 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext {
       sc.removeSparkListener(sparkListener1)
       assert(!sc.listenerBus.listeners.contains(sparkListener1))
       assert(sc.listenerBus.listeners.contains(sparkListener2))
+    } finally {
```

Comment:
Ah, I just remembered: `LocalSparkContext` is actually managing `sc` for every test, including stopping it after it finishes. So this particular change isn't needed, but, the problem is that the `sc` managed for this test is ignored and never closed because it's discarded in the first line. If anything, for tidiness, `sc.stop()` needs to be the first line.